src/lib/journey/_utilities/map-stage.utilities.ts

```diff
@@ -2,14 +2,19 @@ import Generic from '$journey/stages/generic.svelte';
 import Registration from '$journey/stages/registration.svelte';
 import UsernamePassword from '$journey/stages/username-password.svelte';
 
-export function mapStepToStage(currentStep: any) {
+import type { StepTypes } from '$journey/journey.interfaces';
+
+export function mapStepToStage(currentStep: StepTypes) {
   /** *********************************************************************
    * SDK INTEGRATION POINT
    * Summary:SDK step method for getting the stage value
    * ----------------------------------------------------------------------
    * Details: This method is helpful in quickly identifying the stage
    * when you want to provide special layout or handling of the form
    ********************************************************************* */
+  if (!currentStep || currentStep.type !== 'Step') {
+    return Generic;
```

Comment:
What is the meaning of Generic in this instance? I'm a little confused as to what this could mean in a Step

Reply:
You can think of `Generic` in this sense as an unnamed step. So, when there's no `stage` value provided in the step, this `Generic` component is used. When there is a `stage` value, it will use the the named stage component, if matched.

The reason this particular condition is used is prior to the `switch` below is if the app is still booting up, there is no "step" data, so we need to render the `Generic` component as the named stage components have certain assumptions about the step data.